babel/numbers.py

```diff
@@ -317,70 +317,163 @@ def _is_active(start, end):
     return result
 
 
-def get_decimal_symbol(locale: Locale | str | None = LC_NUMERIC) -> str:
+def _get_numbering_system(locale: Locale, numbering_system: Literal["default"] | str = "latn") -> str:
+    if numbering_system == "default":
+        return locale.default_numbering_system
+    else:
+        return numbering_system
+
+
+def _get_number_symbols(
+    locale:  Locale | str | None = LC_NUMERIC,
+    *,
+    numbering_system: Literal["default"] | str = "latn"
+) -> LocaleDataDict:
+    parsed_locale = Locale.parse(locale)
+    numbering_system = _get_numbering_system(parsed_locale, numbering_system)
+    try:
+        return parsed_locale.number_symbols[numbering_system]
+    except KeyError:
+        raise UnsupportedNumberingSystemError(numbering_system, parsed_locale)
+
+
+class UnsupportedNumberingSystemError(Exception):
+    """Exception thrown when an unsupported numbering system is requested for the given Locale."""
+
+    def __init__(self, numbering_system: str, locale: Locale | str | None) -> None:
```

Comment:
This does need to be (and currently is) constructed with an actual Locale.
```suggestion
    def __init__(self, numbering_system: str, locale: Locale) -> None:
```

Reply:
Whole `__init__` function based on the comment above